acstools/acszpt.py

```diff
@@ -276,16 +267,17 @@ def _parse_response(self):
         # the html table. The first 6 elements in the list of all elements
         # correspond to the column names. The values after that correspond to
         # actual data for each row
-        for j in range(6):
-            for i, val in enumerate(td[j::6]):
+        BLOCK_SIZE = 6
+        for j in range(BLOCK_SIZE):
+            for i, val in enumerate(td[j::BLOCK_SIZE]):
```

Comment:
Tsk tsk, double loop. If in the future, this turns out to be a bottleneck, you might need to revise a more efficient way to parse. I'll leave it be for now.

Reply:
I figured out a better way to do it with `numpy` and no loops 😛 . I think I'll make a `dict` to store the units in and then we can just loop over the `dict` to assign units to each column without the need to have the 6 `if/elif` statements. Once that is done we can just combine the `_parse_response()` and `_format_results()` methods into a single method. What do you think?

```console
In [68]: q._submit_request()                                                                                            

In [69]: soup = BeautifulSoup(q._response.read(), 'html.parser')                                                        

In [70]: td = soup.find_all('td')                                                                                       

In [71]: td = [val.text.split(' ')[0] for val in td]                                                                    

In [72]: a = np.reshape(td, (int(len(td)/BLOCK_SIZE), BLOCK_SIZE))                                                      

In [73]: a[0]                                                                                                           
Out[73]: 
array(['FILTER', 'PHOTPLAM', 'PHOTFLAM', 'STmag', 'VEGAmag', 'ABmag'],
      dtype='<U9')

In [74]: a[1]                                                                                                           
Out[74]: 
array(['F220W', '2255.6', '8.166e-18', '21.620', '21.870', '23.546'],
      dtype='<U9')

In [75]: tb = Table(a[1, :], names = a[0], dtype = [str, float, float, float, float, float])

In [76]: tb                                                                                                             
Out[76]: 
<Table length=17>
FILTER PHOTPLAM  PHOTFLAM  STmag  VEGAmag  ABmag 
 str9  float64   float64  float64 float64 float64
------ -------- --------- ------- ------- -------
 F220W   2255.6 8.166e-18   21.62   21.87  23.546
 F250W   2716.1 4.673e-18  22.226  22.255  23.748
 F330W   3362.8  2.22e-18  23.034  22.908  24.093
 F344N   3433.8 2.209e-17   20.54    20.4  21.553
 F435W   4310.8 5.333e-19  24.583  25.185  25.102
 F475W   4775.2 2.893e-19  25.247  25.632  25.544
 F502N   5022.9 8.003e-18  21.642  21.909  21.829
 F550M   5579.7 5.818e-19  24.488  24.417  24.447
 F555W   5355.5 2.995e-19  25.209  25.258  25.257
 F606W   5886.5 1.262e-19  26.147  25.903   25.99
 F625W   6295.0 1.949e-19  25.675  25.206  25.372
 F658N   6583.7 3.333e-18  22.593   21.81  22.193
 F660N   6599.0 8.787e-18   21.54  20.849  21.135
 F775W   7664.7  1.93e-19  25.686  24.566  24.955
 F814W   8114.9  1.27e-19   26.14  24.854  25.286
F850LP   9144.0  2.25e-19   25.52  23.878  24.406
 F892N   8916.1 2.439e-18  22.932  21.379  21.873
```

Reply:
Nice! But is there any chance at this point that the query would return empty table (instead of failing in the checks)? If there is a chance, this would give `IndexError`.